File: backend/app/schemas/conversation.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime

class ConversationCreate(BaseModel):
    # No fields needed for creation, but we can add title if desired
    title: Optional[str] = None

class MessageResponse(BaseModel):
    id: str
    conversation_id: str
    role: str  # 'user' or 'assistant'
    content: str
    created_at: datetime

    model_config = {'from_attributes': True}

class ConversationResponse(BaseModel):
    id: str
    user_id: str
    title: Optional[str]
    created_at: datetime
    updated_at: datetime

    model_config = {'from_attributes': True}

class ConversationListResponse(BaseModel):
    conversations: List[ConversationResponse]

    model_config = {'from_attributes': True}

class ConversationWithMessagesResponse(BaseModel):
    id: str
    user_id: str
    title: Optional[str]
    created_at: datetime
    updated_at: datetime
    messages: List[MessageResponse]

    model_config = {'from_attributes': True}

File: frontend/src/types/conversation.ts
export interface ConversationCreate {
  title?: string;
}

export interface ConversationResponse {
  id: string;
  user_id: string;
  title?: string;
  created_at: string; // ISO string
  updated_at: string; // ISO string
}

export interface ConversationListResponse {
  conversations: ConversationResponse[];
}

export interface ConversationWithMessagesResponse {
  id: string;
  user_id: string;
  title?: string;
  created_at: string; // ISO string
  updated_at: string; // ISO string
  messages: MessageResponse[];
}

export interface MessageResponse {
  id: string;
  conversation_id: string;
  role: 'user' | 'assistant';
  content: string;
  created_at: string; // ISO string
}
